src/engine/context.rs

```diff
@@ -0,0 +1,67 @@
+use crate::error::{bail, Result};
+use crate::parser::ast::*;
+use serde_json::Value;
+
+pub struct Context {
+    #[allow(dead_code)]
+    data: Value,
+}
+
+impl Context {
+    pub fn new(data: Value) -> Context {
+        Context { data }
+    }
+
+    fn lookup_variable_value<'a>(&self, value: &'a Value, identifier: &IdentifierValue) -> Result<&'a Value> {
+        match identifier {
```

Comment:
This match may benefit from extracting functions per match-arm, it's a bit lengthy for me to be able to read and comprehend at once.